### PR TITLE
feat(FX-4027): Add keyword filter to artist page's artwork grid

### PR DESCRIPTION
**Pull request by nickskalkin**
The type of this PR is: **Feature**

This PR solves [FX-4027]

### Description

As a collector
When I visit the artist page’s artwork grid
I see a new keyword filter
And when I enter a keyword
the grid is filtered down to match that query

I've created a feature toggle in Unleash, it's called `artist-artwork-grid-keyword-search` and currently enabled for development and staging

### Demo

* [Demo app](https://artwork-grid-keyword-filter.artsy.net)
* [Banksy grid](https://artwork-grid-keyword-filter.artsy.net/artist/banksy/works-for-sale?sort=-default_trending_score)

https://user-images.githubusercontent.com/3934579/175893281-c8b0e340-644b-4087-9d80-0496e926e4dd.mp4

[FX-4027]: https://artsyproduct.atlassian.net/browse/FX-4027?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mdole — DISMISSED** [review]
@nickskalkin nice work! One high-level thought: since this PR doesn't include support for `keyword` in alert creation, should we put it behind a feature flag until we have a chance to complete that work (ticketed [here](https://artsyproduct.atlassian.net/browse/FX-4029))? 

It feels a bit confusing to be able to filter down to a keyword but not include it in a set alert, so my tendency would be to feature flag it for now, but could also be a good question for Mel.

File: src/v2/Apps/Artist/Routes/WorksForSale/Components/ArtistArtworkFilter.tsx [hunk 1]
```diff
@@ -75,6 +91,24 @@ const ArtistArtworkFilter: React.FC<ArtistArtworkFilterProps> = props => {
...
+      <SizeFilter expanded />
+      <WaysToBuyFilter expanded />
+      <MaterialsFilter />
+      <ArtistNationalityFilter />
+      <ArtworkLocationFilter />
+      <TimePeriodFilter />
+      <ColorFilter />
+      <PartnersFilter />
+    </>
+  )
```
Comment (dimatretyak) [inline src/v2/Apps/Artist/Routes/WorksForSale/Components/ArtistArtworkFilter.tsx:110]: **suggestion**: create a separate component for this

**Review by dimatretyak — CHANGES_REQUESTED** [review]
Unfortunately, there are a couple of problems that need to be fixed for mWeb:
1. Query to Metaphysics should be executed only when the "Show Results" button has been pressed
2. "Clear all" button don't clear the keyboard search input

**Comment by nickskalkin** [comment]
hey @mdole @dimatretyak made changes you suggested, could you take a look?

File: src/v2/Apps/Artist/Routes/WorksForSale/Components/ArtistArtworkFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      <WaysToBuyFilter expanded />
+      <MaterialsFilter />
+      <ArtistNationalityFilter />
+      <ArtworkLocationFilter />
+      <TimePeriodFilter />
+      <ColorFilter />
+      <PartnersFilter />
+    </>
+  )
+}
```
Comment (dimatretyak) [inline src/v2/Apps/Artist/Routes/WorksForSale/Components/ArtistArtworkFilters.tsx:45]: **praise**: thank you for these changes 👏

File: src/v2/Components/ArtworkFilter/ArtworkFilterContext.tsx [hunk 3]
```diff
@@ -20,6 +20,7 @@ export const initialArtworkFilterState: ArtworkFilters = {
   sort: "-decayed_merch",
   artistIDs: [],
   attributionClass: [],
+  keyword: undefined,
```
Comment (dimatretyak) [inline src/v2/Components/ArtworkFilter/ArtworkFilterContext.tsx:23]: **question**: can you please remind me why `keyword` is `undefined` here?

**Review by dimatretyak — APPROVED** [review]

**Pull request merged**